### DISCUSSION TITLE
request: login as android tablet client instead of Windows client

### DISCUSSION THREAD

**Discussion by ha1fstack** [storycraft/node-kakao#152] [Ideas]
Kakaotalk update which supports android tablet multi device was released today.
If node-kakao client virtually login as tablet, it should be much convenient since you can now use real PC client simultaneously.
Configurable client login argument option (whether login as tablet or Windows) would be useful.

**Comment by storycraft** [comment]
I don't know how tablet device login. Connecting loco server may not have big difference if you obtain access token somehow. Customzing client config is already available but may not work if there are difference with pc. Check default configuration from `src/config/client-config.ts`. You can override some config value like this.
```ts
new TalkClient('name', 'deviceUUID', {
    'agent': 'android'
});
```
^ This doesn't work due to difference between mobile and pc.

Note: NodeKakao v4 complete rewrite is currently being actively developed. Check dev branch for more information. You may request as feature when pull request draft is available.

**Comment by storycraft** [comment]
From v4, to login as android tablet, you should change several config values.

Sample code below
```typescript
import { util, api, AuthApiClient, TalkClient } from 'node-kakao';

const config = {
  agent: 'android',

  // Network operator (999 if unknown)
  mccmnc: '999',

  // This should be valid tablet device model or server will refuse request.
  deviceModel: 'SM-T976N',

  // PC version append build number in appVersion, android uses only version number.
  appVersion: '9.2.1',
  version: '9.2.1',

  // 0 == WIFI, 3 == Cellular
  netType: 0,

  subDevice: true
}

// Generate random device uuid.
// If you specify already registered pc uuid, you will login as pc.
const tabletUUID = util.randomAndroidSubDeviceUUID();

// Override default config,  uses AndroidSubXVCProvider for xvc calculating.
const api = await AuthApiClient.create('NAME', tabletUUID, config, api.xvc.AndroidSubXVCProvider);
const client = new TalkClient(config);
```

**Comment by ghost** [comment]
![tenor](https://user-images.githubusercontent.com/78786016/108171457-e17e6480-713e-11eb-85ce-36fa69dd74c9.gif)

**Comment by ghost** [comment]
4.0.0-rc.2에서 태블릿으로 로그인을 해본 결과 로그인에 성공하였다고 나오나 대부분의 이벤트가 오지 않고 얼마 뒤에 아무런 에러도 없이 연결이 끊깁니다. 🤔 

확인결과 문법상 오류로 확인

**Comment by storycraft** [comment]
해당 문제는 로그, 오류 내용과 함께 이슈를 작성해주세요

**Comment by ghost** [comment]
> 해당 문제는 로그, 오류 내용과 함께 이슈를 작성해주세요

#149
